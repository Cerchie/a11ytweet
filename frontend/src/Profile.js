import * as React from 'react'
import { useState, useContext } from 'react'
import UserContext from './UserContext'
import { useHistory } from 'react-router-dom'

function Profile({ saveProfile }) {
    if (useContext(UserContext) !== undefined) {
        const { currentUser } = useContext(UserContext)

        const [formData, setFormData] = useState({
            username: currentUser.username,
            password: '',
        })
        const [formErrors, setFormErrors] = useState([])
        const [saveConfirmed, setSaveConfirmed] = useState(false)

        console.debug(
            'ProfileForm',
            'currentUser=',
            currentUser,
            'formData=',
            formData,
            'formErrors=',
            formErrors,
            'saveConfirmed=',
            saveConfirmed
        )
        function goHome() {
            let history = useHistory()
            history.push('/gohome')
        }
        async function handleSubmit(evt) {
            evt.preventDefault()

            let profileData = {
                password: formData.password,
            }
            let username = formData.username
            let updatedUser

            try {
                updatedUser = await saveProfile(username, profileData)
            } catch (errors) {
                setFormErrors(errors)
                return
            }
            setFormData((f) => ({ ...f, password: '' }))
            setFormErrors([])
            setSaveConfirmed(true)
        }

        function handleChange(evt) {
            const { name, value } = evt.target
            setFormData((f) => ({
                ...f,
                [name]: value,
            }))
            setFormErrors([])
        }
        //maybe add a 'if Curruser...''
        return (
            <form onSubmit={handleSubmit}>
                <>
                    <label>Username</label>
                    <input
                        name="username"
                        className="form-control"
                        value={formData.username}
                        onChange={handleChange}
                    />
                </>
                <>
                    <label>Password</label>
                    <input
                        type="password"
                        name="password"
                        className="form-control"
                        value={formData.password}
                        onChange={handleChange}
                    />
                </>
                {saveConfirmed ? goHome() : null}
                <button
                    type="submit"
                    className="btn btn-primary float-right"
                    onSubmit={handleSubmit}
                >
                    Submit
                </button>
            </form>
        )
    }
    return <> No Profile </>
}

export default Profile
