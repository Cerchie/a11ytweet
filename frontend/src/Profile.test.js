import { render, screen } from '@testing-library/react'
import React from 'react'
import Profile from './Profile'

test('renders comp', () => {
    render(<Profile />)
})
