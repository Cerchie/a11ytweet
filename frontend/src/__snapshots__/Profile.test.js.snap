// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <form>
    <div
      class="form-group"
    >
      <label>
        Username
      </label>
      <input
        class="form-control"
        name="username"
        value="testuser"
      />
    </div>
    <div
      class="form-group"
    >
      <label>
        Password
      </label>
      <input
        class="form-control"
        name="password"
        type="password"
        value=""
      />
    </div>
    <button
      class="btn btn-primary float-right"
      type="submit"
    >
      Submit
    </button>
  </form>
</DocumentFragment>
`;
